boto3/resources/params.py

```diff
@@ -43,20 +43,25 @@ def create_request_parameters(parent, request_model, params=None):
 
     for param in request_model.params:
         source = param.source
-        source_type = param.source_type
         target = param.target
 
-        if source_type in ['identifier', 'dataMember']:
+        if source == 'identifier':
             # Resource identifier, e.g. queue.url
+            value = getattr(parent, xform_name(param.name))
+        elif source == 'data':
             # If this is a dataMember then it may incur a load
             # action before returning the value.
-            value = getattr(parent, xform_name(source))
-        elif source_type in ['string', 'integer', 'boolean']:
+            # TODO: This should be a JMESPath query on the parent
```

Comment:
This parameter/identifier's source type is `data`, which means it comes from a data member on the associated parent resource. Data members are accessed by `path`, which is a JMESPath query. Once JMESPath supports querying attributes of objects then this line should change from a `getattr` call to a `jmespath.search` or something along those lines. For now it works in most cases since `path` is just the name of an attribute (e.g. `VpcId`), but the spec allows for any JMESPath (e.g. `Items[0].id`).